src/web/components/ClientSideTokenGeneration/CstgAddDomainDialog.spec.tsx

```diff
@@ -86,17 +77,14 @@ describe('CstgDomainAddDomainDialog', () => {
       />
     );
 
-    await user.type(screen.getByRole('textbox', { name: 'newDomains' }), 'test');
-
     await submitDialog();
 
     expect(screen.getByRole('alert')).toBeInTheDocument();
```

Comment:
I believe this test isn't actually testing the right case here, the alert it is producing seems to be the empty input case, rather than the invalid domain case. i.e. If I add this line: `expect(screen.getByText('Please specify domains.')).toBeInTheDocument();` it still passes. 

I think we need to do something similar to what we did in the happy case, i.e. use `user.type(...)` in order to set this up properly - perhaps the `onAddDomainsMock` is confusing.

If we do that, then we should be able to assert on the correct error text.

So I think this test should become:
```
it('should render error when user types single incorrect domain', async () => {
  const user = userEvent.setup();

  const onAddDomainsMock = jest.fn(() => {
    return Promise.resolve(['newDomains']);
  });

  render(
    <CstgAddDomainDialog
      onAddDomains={onAddDomainsMock}
      onOpenChange={() => {}}
      existingDomains={[]}
    />
  );

  await user.type(screen.getByRole('textbox', { name: 'newDomains' }), 'newDomains');

  await submitDialog();

  expect(
    screen.getByText('The domains entered are invalid root-level domains', { exact: false })
  ).toBeInTheDocument();
});
```

Reply:
I think this also applies to the below test: `should render error if user types in at least one incorrect domain in a list`